modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java

```diff
@@ -95,12 +95,16 @@ private HybridTimestamp(long time) {
     }
 
     /**
-     * Converts primitive {@code long} representation into a hybrid timestamp instance.
-     * {@link #NULL_HYBRID_TIMESTAMP} is interpreted as {@code null}.
+     * Converts {@link Long} representation into a hybrid timestamp instance.
+     * {@link #NULL_HYBRID_TIMESTAMP} and {@code null} are interpreted as {@code null}.
      *
      * @throws IllegalArgumentException If timestamp is negative.
      */
-    public static @Nullable HybridTimestamp nullableHybridTimestamp(long time) {
+    public static @Nullable HybridTimestamp nullableHybridTimestamp(@Nullable Long time) {
```

Comment:
There are dozen of places where we propagate long and not Long - I'd rather try to avoid excess unboxing. Besides that HybridTimestamp with this change is no longer symmetrical: longValue(), getPhysical() and others return long.